src/ipc/ipc4/handler.c

```diff
@@ -568,7 +568,7 @@ static int ipc4_set_pipeline_state(struct ipc4_message_request *ipc4)
 
 	/* Run the prepare phase on the pipelines */
 	for (i = 0; i < ppl_count; i++) {
-		ppl_icd = ipc_get_comp_by_ppl_id(ipc, COMP_TYPE_PIPELINE, ppl_id[i]);
+		ppl_icd = ipc_get_comp_by_ppl_id(ipc, COMP_TYPE_PIPELINE, ppl_id[i], true);
```

Comment:
Hmm, boolean args have an impact to readability of the calling code. Given this is used in a lot of places, I'd prefer to add named defined/enum (IPC_COMP_ALL and IPC_COMP_IGNORE_REMOTE) and then have more readable code at the calling function:

```
ipc_get_comp_by_ppl_id(ipc, COMP_TYPE_PIPELINE, ppl_id[i], IPC_COMP_IGNORE_REMOTE);`

ipc_get_comp_by_ppl_id(ipc, COMP_TYPE_PIPELINE, ppl_id[i], IPC_COMP_ALL);
```

Reply:
Changed according to Kai suggestion.